hadoop-hdds/server-scm/src/main/java/org/apache/hadoop/hdds/scm/container/placement/algorithms/SCMContainerPlacementRackScatter.java

```diff
@@ -143,10 +147,12 @@ public List<DatanodeDetails> chooseDatanodes(
 
     List<Node> toChooseRacks = new LinkedList<>();
     Set<DatanodeDetails> chosenNodes = new LinkedHashSet<>();
-    List<Node> unavailableNodes = new ArrayList<>();
     Set<Node> skippedRacks = new HashSet<>();
     if (excludedNodes != null) {
-      unavailableNodes.addAll(excludedNodes);
+      unavailableNodes.addAll(
+              excludedNodes.stream()
+                      .filter(node->!unavailableNodes.contains(node))
+                      .collect(Collectors.toList()));
     }
```

Comment:
Since neither `excludedNodes`, nor `unavailableNodes` changes after filling `unavailableNodes` initially, is this second `addAll` call necessary?

Reply:
I am just adding the nodes which might not have come as part of the available nodes list.

Reply:
Oh, I see.